### PR TITLE
[incubator-kie-issues#1719] Enable onEntry/onExit scripts in embedded nodes

### PR DESCRIPTION
**Pull request by martinweiler**
Closes https://github.com/apache/incubator-kie-issues/issues/1719

### CODE REVIEW

**Comment by kie-ci3** [comment]
**PR job** `#1` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3815 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3815/1/display/redirect

**Test results:**
- PASSED: 3515
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3815/1/testReport/org.jbpm.bpmn2/IntermediateEventTest/testMultiInstanceLoopCharacteristicsTaskWithOutputCmpCondSequential/">org.jbpm.bpmn2.IntermediateEventTest.testMultiInstanceLoopCharacteristicsTaskWithOutputCmpCondSequential</a></summary>
<br/>Expected size: 1 but was: 0 in:<br/>[]
</details>

**Review by yesamer — APPROVED** [review]

**Review by jomarko — COMMENTED** [review]
@martinweiler thank you for the PR. I am always amazed and inspired by the amount of work you are able to deliver.

I have few comments inline and one general question. Is the user task an embedded node? Asking as the only changed node in the PR is a human task.  While I thought in BPMN 
 the embedded word is always used for embedded subprocesses. Sorry if the question is silly and just shows my missing knowledge.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/FlowTest.java [hunk 1]
```diff
@@ -857,30 +857,32 @@ public void beforeNodeTriggered(ProcessNodeTriggeredEvent event) {
...
-        kruntime.getKogitoWorkItemManager().registerWorkItemHandler("testWI", handler);
-        kruntime.getKogitoWorkItemManager().registerWorkItemHandler("testWI2", handler2);
-        Map<String, Object> params = new HashMap<>();
-        params.put("x", -1);
-        KogitoProcessInstance processInstance = kruntime.startProcess("Process_1", params);
+        ProcessTestHelper.registerHandler(app, "testWI", handler);
+        ProcessTestHelper.registerHandler(app, "testWI2", handler2);
 
-        assertProcessInstanceActive(processInstance);
+        Process<InclusiveGatewayWithLoopInsideSubprocessModel> process = InclusiveGatewayWithLoopInsideSubprocessProcess.newProcess(app);
```
[the remark this answers is not in the document]
Comment (pefernan) [reply jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/FlowTest.java:866]: @jomarko that`InclusiveGatewayWithLoopInsideSubprocessModel ` is generated during the build, if you run a mvn clean package locally you'll find them in the project target folder under the `generated-sources` folder

**Review by pefernan — APPROVED** [review]

File: jbpm/jbpm-tests/src/test/bpmn/org/jbpm/bpmn2/loop/BPMN2-MultiInstanceLoopCharacteristicsTaskWithOutputCmpCondSequential.bpmn2 [hunk 2]
```diff
@@ -45,8 +45,8 @@
     <startEvent id="_1" name="StartProcess" />
     <userTask id="_2" name="Hello" tns:taskName="Human Task">
       <extensionElements>
-        <tns:onEntry-script>
-          <tns:script>System.out.println("Hello");context.setVariable("itemOut", "test");</tns:script>
+        <tns:onEntry-script scriptFormat="http://www.java.com/java">
+          <tns:script>System.out.println("Hello");kcontext.setVariable("itemOut", "test");</tns:script>
```
[the remark this answers is not in the document]
Comment (martinweiler) [reply jbpm/jbpm-tests/src/test/bpmn/org/jbpm/bpmn2/loop/BPMN2-MultiInstanceLoopCharacteristicsTaskWithOutputCmpCondSequential.bpmn2:49]: I think you have a valid point here. Those messages are just adding noise to the log files, and don't provide any real value. I have created https://github.com/apache/incubator-kie-issues/issues/1726 now to look into this some more. This affects not only this particular bpmn file.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/FlowTest.java [hunk 3]
```diff
@@ -837,12 +839,10 @@ public void beforeNodeTriggered(ProcessNodeTriggeredEvent event) {
     }
 
     @Test
-    @Disabled("On Exit not supported, see https://issues.redhat.com/browse/KOGITO-2067")
```
[the remark this answers is not in the document]
Comment (martinweiler) [reply jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/FlowTest.java:849]: Yes, we should, and will. But some of them require additional fixes in the engine, so we will have to tackle them one by one.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/FlowTest.java [hunk 4]
```diff
@@ -857,30 +857,32 @@ public void beforeNodeTriggered(ProcessNodeTriggeredEvent event) {
...
-
-        assertProcessInstanceActive(processInstance);
-        kruntime.getKogitoWorkItemManager().completeWorkItem(handler2.getWorkItem().getStringId(), null);
-        assertProcessInstanceActive(processInstance);
-        kruntime.getKogitoWorkItemManager().completeWorkItem(handler2.getWorkItem().getStringId(), null);
-        assertProcessInstanceActive(processInstance);
-        kruntime.getKogitoWorkItemManager().completeWorkItem(handler.getWorkItem().getStringId(), null);
-        assertProcessInstanceCompleted(processInstance);
+        assertThat(instance.status()).isEqualTo(ProcessInstance.STATE_ACTIVE);
+        instance.completeWorkItem(handler2.getWorkItem().getStringId(), null);//After this line, state of the process instance is error state instead of active
```
[the remark this answers is not in the document]
Comment (martinweiler) [reply jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/FlowTest.java]: good catch, thanks! comment removed now

**Comment by martinweiler** [comment]
@jomarko thanks for looking into this issue, and for your input. This is very much appreciated, as you bring in some new perspectives and raise valid points!

Regarding your question on embedded nodes and user tasks, what is used in some of the tests here are multi-instance tasks (similar to MI subprocesses, but just in a single task). For the engine, this task is then embedded in a ForEachNode. I hope this helps to clarify.

**Pull request merged**